### DISCUSSION TITLE
A way to detect burned/ broken components

### DISCUSSION THREAD

**Discussion by Angelo50** [SpiceSharp/SpiceSharp#175] [General]
Hi,

Sorry for this question but I did not find any answer anywhere
I 'm working on an unity  project using your greatfull nuggets
But Is there a way to detect if for exemple the breakdown voltage of a diode has been reached or do I have to handle myself with logic ?

Thanks

Angelo.

**Comment by svenboulanger** [comment]
Hello,

There are two ways to go about this.
1. The first is to just check manually when exporting simulation data. You would check if the voltage across diodes cross their breakdown voltage and act accordingly.
2. If you want to make use of something a bit more involved, I suggest working from the existing diode models and adding perhaps an event, a flag, etc. to signal that a diode has gone into breakdown region. Needless to say that this is going to be a bit more involved as you'd have to recreate all the behaviors appropriately.

I would recommend option 1 if it just checking one or the other. You can find the breakdown voltage of all diodes in the circuit by searching the parameters in the behavior, and compare the voltage across the diode using exports.

**Comment by Angelo50** [comment]
Thank you very much for your answer

Angelo.
